server/api/src/rsGetHostForGet.cpp

```diff
@@ -17,8 +17,73 @@
 #include "irods/rsGetHostForGet.hpp"
 #include "irods/irods_resource_backport.hpp"
 #include "irods/irods_resource_redirect.hpp"
+#include "irods/irods_at_scope_exit.hpp"
 
+#include <nlohmann/json.hpp>
+
+#include <set>
 #include <cstring>
+#include <optional>
+
+int rs_get_resource_info_for_operation(
+    rsComm_t*     rsComm,
+    dataObjInp_t* dataObjInp,
+    char **       outInfo )
+{
+    rodsServerHost_t *rodsServerHost;
+    const int remoteFlag = getAndConnRemoteZone(rsComm, dataObjInp, &rodsServerHost, REMOTE_OPEN);
+    if (remoteFlag < 0) {
+        return remoteFlag;
+    }
+    else if (REMOTE_HOST == remoteFlag) {
+        const int status = rc_get_resource_info_for_operation(rodsServerHost->conn, dataObjInp, outInfo);
+        if (status < 0) {
+            return status;
+        }
+    }
+    else {
+        std::string location {};
+        // =-=-=-=-=-=-=-
+        // working on the "home zone", determine if we need to redirect to a different
+        // server in this zone for this operation.  if there is a RESC_HIER_STR_KW then
+        // we know that the redirection decision has already been made
+        std::string hier{};
+        const char * op_type = getValByKey(&dataObjInp->condInput, GET_RESOURCE_INFO_OP_TYPE_KW);
+        std::set<std::string> allowed_ops{ irods::CREATE_OPERATION,
+                                           irods::WRITE_OPERATION,
+                                           irods::OPEN_OPERATION  };
+        if (!op_type || std::find(allowed_ops.begin(), allowed_ops.end(), op_type) == allowed_ops.end()) {
+            return RE_UNSUPPORTED_OP_OR_TYPE;
+        }
+        const std::string operation{op_type};
```

Comment:
Could we use the result of `std::find` to avoid making a copy here?

Reply:
That particular expert use of the algorithm presently "has above-head trajectory...."

Reply:
I think I know what you refer to @alanking, I could instantiate an iterator and reference the string object through that, but I think the result would be less readable.  That said, I'm willing to have someone show me how it should look : )

Reply:
I gave this a stab, and still don't think avoiding the string copy is worth the extra unreadability.  Please advise....

Reply:
https://github.com/d-w-moore/irods/blob/6ace74d42fcc9be56aa145de041e0b913f30fd20/server/api/src/rs_get_resource_info_for_operation.cpp#L55

Reply:
done